pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/FlinkSegmentWriter.java

```diff
@@ -124,12 +132,26 @@ public void init(TableConfig tableConfig, Schema schema, Map<String, String> bat
         "batchConfigMaps must contain only 1 BatchConfig for table: %s", _tableNameWithType);
 
     Map<String, String> batchConfigMap = _batchIngestionConfig.getBatchConfigMaps().get(0);
+    batchConfigMap.put(BatchConfigProperties.PARTITION_ID, Integer.toString(_indexOfSubtask));
+    batchConfigMap.put(BatchConfigProperties.SEQUENCE_ID, Integer.toString(_seqId));
```

Comment:
Will this not always put the sequence-id as 0 in all scenarios? Can this cause conflicts when creation time is same as upsert table might end up replacing the old segment?

Reply:
Thanks for catching this, it needs to be in flush as FlinkSegmentWriter gets initialised once. I'll add an integration test around this as well.

Reply:
Fixed. Adding integration test caught another pre-existing bug with flush happening at flushThreshold+1. Fixed as part of the same PR.